libs/SalesforceSDKCore/SalesforceSDKCoreTests/SalesforceRestAPITests.m

```diff
@@ -2465,6 +2465,22 @@ - (SFOAuthCredentials *)getTestCredentialsWithDomain:(nonnull NSString *)domain
     return creds;
 }
 
+- (void)testRedirect {
+    // Create contact to fetch image for
+    NSDictionary *fields = @{FIRST_NAME: @"John", LAST_NAME: [self generateRecordName]};
+    SFRestRequest *contactRequest = [[SFRestAPI sharedInstance] requestForCreateWithObjectType:CONTACT fields:fields apiVersion:kSFRestDefaultAPIVersion];
+    SFNativeRestRequestListener *contactListener = [self sendSyncRequest:contactRequest];
+    NSString *contactId = ((NSDictionary *)contactListener.dataResponse)[LID];
+
+    // Authenticated request for contact image, shouldn't automatically redirect
+    NSString *path = [NSString stringWithFormat:@"/services/images/photo/%@", contactId];
+    SFRestRequest *request = [SFRestRequest requestWithMethod:SFRestMethodGET path:path queryParams:nil];
+    request.endpoint = @"";
+    SFNativeRestRequestListener *listener = [self sendSyncRequest:request];
+    NSInteger statusCode = [(NSHTTPURLResponse *)listener.rawResponse statusCode];
+    XCTAssertEqual(statusCode, 302, @"Request did not return 302");
```

Comment:
Let's add a line here to read the URL that it's going to and assert it's not `nil`.